homeassistant/components/tfiac/climate.py

```diff
@@ -0,0 +1,188 @@
+"""Climate platform that offers a climate device for the TFIAC protocol."""
+from concurrent import futures
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.climate import PLATFORM_SCHEMA, ClimateDevice
+from homeassistant.components.climate.const import (
+    STATE_AUTO, STATE_COOL, STATE_DRY, STATE_FAN_ONLY, STATE_HEAT,
+    SUPPORT_FAN_MODE, SUPPORT_ON_OFF, SUPPORT_OPERATION_MODE,
+    SUPPORT_SWING_MODE, SUPPORT_TARGET_TEMPERATURE)
+from homeassistant.const import ATTR_TEMPERATURE, CONF_HOST, TEMP_FAHRENHEIT
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+
+DOMAIN = 'tfiac'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+REQUIREMENTS = ['pytfiac==0.3']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TEMP = 61
+MAX_TEMP = 88
+OPERATION_MAP = {
+    STATE_HEAT: 'heat',
+    STATE_AUTO: 'selfFeel',
+    STATE_DRY: 'dehumi',
+    STATE_FAN_ONLY: 'fan',
+    STATE_COOL: 'cool',
+}
+OPERATION_MAP_REV = {
+    v: k for k, v in OPERATION_MAP.items()}
+FAN_LIST = ['Auto', 'Low', 'Middle', 'High']
+SWING_LIST = [
+    'Off',
+    'Vertical',
+    'Horizontal',
+    'Both',
+]
+
+CURR_TEMP = 'current_temp'
+TARGET_TEMP = 'target_temp'
+OPERATION_MODE = 'operation'
+FAN_MODE = 'fan_mode'
+SWING_MODE = 'swing_mode'
+ON_MODE = 'is_on'
+
+
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
+    """Set up the TFIAC climate device."""
+    from pytfiac import Tfiac
+
+    tfiac_client = Tfiac(config[CONF_HOST])
+    try:
+        await tfiac_client.update()
+    except futures.TimeoutError:
+        return
+    async_add_devices([TfiacClimate(hass, tfiac_client)])
+
+
+class TfiacClimate(ClimateDevice):
+    """TFIAC class."""
+
+    def __init__(self, hass, client):
+        """Init class."""
+        self._client = client
+        self._available = True
+
+    @property
+    def available(self):
+        """Return if the device is available."""
+        return self._available
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
```

Comment:
Do we need the throttle or is it enough with setting `SCAN_INTERVAL` constant to this interval?

The throttle will only limit updates per entity and not per home assistant.